awscli/customizations/s3/filegenerator.py

```diff
@@ -208,13 +207,19 @@ def list_files(self, path, dir_op):
                         for x in self.list_files(file_path, dir_op):
                             yield x
                     else:
-                        size, last_update = get_file_stat(file_path)
-                        last_update = self._validate_update_time(
-                            last_update, path)
-                        yield (
-                            file_path,
-                            {'Size': size, 'LastModified': last_update}
-                        )
+                        stats = self._safely_get_file_stats(path, file_name=name)
```

Comment:
No, you'll note that both the original implementation and my new one use the original `path`.